### PR TITLE
a-tiny-JS-world-done by LGeorge007

### PR DESCRIPTION
**Pull request by LGeorge007**
_Have your app published on [GitHub pages](https://github.com/kottans/frontend-2021-homeworks/blob/main/publish-your-app.md). Make sure your PR title contains the task name. Change task name and links below as appropriate. Remove this paragraph._

### a-tiny-JS-world (pre OOP)

[Demo](https://LGeorge007.github.io/a-tiny-JS-world)
[Code base](https://github.com/LGeorge007/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/LGeorge007/a-tiny-JS-world/index.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+   legs: 2,
+   hands: 2,
+   gender: "male",
+   friends: ["Vladimir", "Mariya", "Oksana"],
+   saying: "Hello!",
+};
+
+const message = function(inhabitant) {
+   return `I'm a ${inhabitant.species}. My name is ${inhabitant.name}. My gender is ${inhabitant.gender}. ` + ((inhabitant.hands !== 0)? `I have ${inhabitant.hands} hands and ${inhabitant.legs} legs. ` : `I have ${inhabitant.legs} paws. `) + `Also I have a friends: ${inhabitant.friends.join(", ")}. ${inhabitant.saying}`;
+};
```
Comment (OleksiyRudenko) [inline submissions/LGeorge007/a-tiny-JS-world/index.js:63]: Let's make this fragment more readable and extensible (imagine we want to add a couple more props)

Check the example below:
```

const someVar = condition ? 'option1' : 'option2';

return [
  `I'm a ${inhabitant.species}`,
  `My name is ${inhabitant.name}`,
  inhabitant.hands 
    ? `${inhabitant.hands} hands and ${inhabitant.legs} legs`
    : `${inhabitant.legs} paws`,
  `more text with a ${someVar}`,
].join('. ');
```

Besides, there is quite a repetition of `inhabitant`. How about decomposition?

```
const longerStringLiteral = `${obj.propA} some text ${obj.propB}`;

const { propA, propB } = object;

const shorterStringLiteral = `${propA} some text ${propB}`;
```

Do not copypaste the examples but comprehend those and use to your benefit.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@LGeorge007 well done!
Let's improve readability using tricks with `Array` methods.

File: submissions/LGeorge007/a-tiny-JS-world/index.js (hunk 1 above)
Comment (LGeorge007) [reply submissions/LGeorge007/a-tiny-JS-world/index.js]: Thank you!
Your code is really more convenient and prettier!
I've figured out destruction (ES6) in practice.
I didn't know that you can use conditions directly in the array. I forgot that if (0) === false.
Thanks for your prompt.

Please, review.

File: submissions/LGeorge007/a-tiny-JS-world/index.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   friends: ["Vladimir", "Mariya", "Oksana"],
+   saying: "Hello!"
+};
+
+const message = function(inhabitant) {
+   
+   const {species, name, gender, hands, legs, friends, saying} = inhabitant;
+   
+   return [
+      `${saying}`,
```
Comment (OleksiyRudenko) [inline submissions/LGeorge007/a-tiny-JS-world/index.js:66]: @LGeorge007 can you please explain why do you refer a variable in this particular way here?

Comment (LGeorge007) [reply submissions/LGeorge007/a-tiny-JS-world/index.js:66]: Yes. It really doesn't make sense. Why interpolate when there is only one value? When I was writing code at night, it seemed logical to me to bring everything to one form. Sorry. I 'll repare it.

**Review by OleksiyRudenko — APPROVED** [review]
@LGeorge007 great job!

**Pull request merged**